src/usvfs_dll/hookcontext.cpp

```diff
@@ -216,6 +216,56 @@ BOOL HookContext::executableBlacklisted(LPCWSTR wapp, LPCWSTR wcmd) const
   return m_Parameters->executableBlacklisted(app, cmd);
 }
 
+void usvfs::HookContext::addSkipFileSuffix(const std::wstring& fileSuffix) 
+{
+  const auto fsuffix = shared::string_cast<std::string>(fileSuffix, shared::CodePage::UTF8);
+
+  if (fsuffix.empty()) {
+    return;
+  }
+
+  spdlog::get("usvfs")->debug("added skip file suffix '{}'", fsuffix);
+  m_Parameters->addSkipFileSuffix(fsuffix);
+}
+
+void usvfs::HookContext::clearSkipFileSuffixes() 
+{
+  spdlog::get("usvfs")->debug("clearing skip file suffixes");
+  m_Parameters->clearSkipFileSuffixes();
+}
+
+BOOL usvfs::HookContext::fileShouldBeSkipped(const std::wstring& wFilename) const
+{
+  const std::string filename = ush::string_cast<std::string>(wFilename, ush::CodePage::UTF8);
+
+  return m_Parameters->fileShouldBeSkipped(filename);
+}
+
+void usvfs::HookContext::addSkipDirectory(const std::wstring& directory) 
+{
+  const auto dir = shared::string_cast<std::string>(directory, shared::CodePage::UTF8);
```

Comment:
Why do we convert everything to std::string? Could we keep everything in `std::wstring` ?
Haven't been dealing with C++ string types for a bit so not sure

Reply:
I'm quite honestly not sure, I wanted to follow other usvfs code and assumed there's a reason for it. 

DLL exposed functions must take in a `wstring` or `LPCWSTR`, `LPCWSTR` is utf-16 and the gaps my brain has filled in is that we need to cast to utf-8 so that we can do system-compatible string comparisons (since not all systems will use utf-16 for filenames/directory names like Windows does)

Reply:
Rubber ducking here, it looks like the file name pulled from the winapi that's passed into the file/directory functions to check if we should skip it all pass in a `wstring`, so what I said above is completely wrong.

`wstring` could be passed directly to all of these functions, but eventually there does need to be a cast to utf8 since the shared memory implementation of string is used heavily in usvfs, https://github.com/ModOrganizer2/usvfs/blob/master/src/shared/shared_memory.h#L50